Usbipd/UsbIds.cs

```diff
@@ -0,0 +1,340 @@
+﻿//#define USBIDS_DBG_VERBOSE
+#define USBIDS_STOPWATCH
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text.RegularExpressions;
+
+namespace Usbipd
+{
+    /// <summary>
+    /// A C# port of https://github.com/cezanne/usbip-win/blob/master/userspace/lib/names.c (GPLv2+)
+    /// </summary>
+    class UsbIds
+    {
+        static UsbIds()
+        {
+            Load();
+        }
+
+        class UsbVendor
+        {
+            public UsbVendor(string name, uint vendorid)
+            {
+                this.name = name;
+                this.vendorid = vendorid;
+            }
+
+            public string name { get; private set; }
+            public uint vendorid { get; private set; }
+            public Dictionary<uint, UsbProduct> products = new Dictionary<uint, UsbProduct>();
+        }
+
+        class UsbProduct
+        {
+            public uint vendorid { get; private set; }
```

Comment:
C# has `record` for this. (mentioned once, but true for all these nested classes.